client/data/disputes/resolvers.js

```diff
@@ -38,11 +39,18 @@ const formatQueryFilters = ( query ) => ( {
 } );
 
 export function getDisputesCSV( query ) {
+	const queryWithSearch = {
+		...query,
+		search:
+			query.filter === 'awaiting_response'
+				? disputeAwaitingResponseStatuses
+				: query.search,
+	};
+
```

Comment:
💭 it might be better for consistency to apply this in a single place for all `getDisputes`/ "dispute list" API requests – I think `formatQueryFilters` in this file would be suitable since it is also where we are modifying the query in other ways.

```diff
const formatQueryFilters = ( query ) => ( {
	user_email: query.userEmail,
	match: query.match,
	store_currency_is: query.storeCurrencyIs,
	date_before: formatDateValue( query.dateBefore, true ),
	date_after: formatDateValue( query.dateAfter ),
	date_between: query.dateBetween && [
		formatDateValue( query.dateBetween[ 0 ] ),
		formatDateValue( query.dateBetween[ 1 ], true ),
	],
-	search: query.search,
+ // If `awaiting_response` filter is present, use the `search` query to select only those disputes awaiting a response.
+ search:
+		query.filter === 'awaiting_response'
+			? disputeAwaitingResponseStatuses
+			: query.search,
	status_is: query.statusIs,
	status_is_not: query.statusIsNot,
	locale: query.locale,
} );
```

This would then mean we can apply the same filtering to all requests, reducing the chance of inconsistency bugs like the one we're solving in this PR.

We implemented (or really, I did, if looking at `git blame`) the following `awaiting_response` filtering in the `useDisputes` hook, which can be removed if we apply this logic across to all "dispute list" requests. We'd have to do some extra testing to make sure we don't break those endpoints, but it would be worth it IMO.

https://github.com/Automattic/woocommerce-payments/blob/a89f43447c41e2ac74c018357019523f1a024c36/client/data/disputes/hooks.ts#L101-L104

Reply:
Not to say your solution doesn't work, by the way! But it will mean we are applying this logic in different places for different endpoints (`/disputes/`, `/disputes/download`, `/disputes/summary`).

Reply:
Thanks for the review @Jinksi, I've consolidated the code into `formatQueryFilters`